### PR TITLE
feature: add Polygon support

### LINKED ISSUE #853: Add Polygon support
Adding Polygon to Swapr includes:

SDK:
 - [x] add basic config (chainId, Matic native currency, core tokens on Polygon like WETH, USDC, USDT, etc.)
 - [x] enable Sushiswap on Polygon
 - [x] add Quickswap to `UniswapV2RoutablePlatforms`

FE:
 - [x] enable adding/switching network to Polygon
 - [x] disable Swapr specific features when connected to Polygon (disable `Liquidity`, `Rewards`)

Bridge:
 - [ ] enable Polygon support on Socket
 - [ ] add bridgable tokens for Polygon

~Due to upcoming major changes in SDK, #616 is used as a base branch (both on SDK and FE)~.

### PR DESCRIPTION
**Pull request by denciu**
# Summary

Closes #853

- enabled selecting Polygon (connectors, bridge inputs, etc)
- refactored header menu components to be cleaner
- disabled features that require SWPR token (and/or its infrastructure) when connected to a non-SWPR network
- added Quickswap dex 

To test:
- [ ] Polygon is visible and selectable in the Network Switcher
- [ ] `Liquidity` and `Rewards` are not disabled and can be clicked in the menu
- [ ] user is able to connect to polygon (RPC endpoint is correct)

After connectin to Polygon:
- [ ] default currency is `MATIC` and balances are displayed correctly
- [ ] `Liquidity` and `Rewards` are grayed out in the menu and can't be clicked. Badge 'Not available' should be visible 
- [ ] Going to https://swapr.eth.link/#/pools or https://swapr.eth.link/#/rewards redirects to the landing page (routing is disabled for non-SWPR features) 
- [ ] Quickswap and Sushiswap are visible on ecoRouter when some common pair is selected

### CODE REVIEW

File: src/constants/index.tsx [hunk 1]
```diff
@@ -365,6 +383,9 @@ export const NETWORK_OPTIONAL_DETAIL: { [chainId: number]: NetworkOptionalDetail
     partnerChainId: ChainId.ARBITRUM_RINKEBY,
     isArbitrum: false
   }
+  // [ChainId.POLYGON]: {
+  //   isArbitrum: false
+  // }
```
Comment (jorgedclopes) [inline src/constants/index.tsx:388]: We shouldn't have commented code.

**Comment by jorgedclopes** [comment]
Shouldn't we change the version in package?

**Comment by niemam29** [comment]
Not sure if it comes from this pr but shouldn't this inscription be lower?
cach2
Modal from dev:
cach2

**Comment by niemam29** [comment]
### Retested and issues that i mentioned above are now working, but there are few more:
#### On bridge the polygon network is not clickable (that may be on purpose, not sure) and the gnosis chain is in the coming soon section
cach2
#### Another problem is that by entering empty url swap is not displayed by default when user is connected to mainet
cach2

**Comment by kingalg** [comment]
There is a but in token search (I believe that we had something like that before).

If you type "MA" in search "MATIC" appears (and it's correct behaviour)
Screenshot 2022-04-28 at 14 59 08
But if you try search from "the middle" for example typing "TIC" MATIC will not appear (but wrapped MATIC will...)
Screenshot 2022-04-28 at 15 00 34
That's quite strange behaviour.

**Comment by MilanVojnovic95** [comment]
**Before connection to Polygon:**

 Polygon is visible but **is not selectable** in the Network Switcher ❌ 
 the same situation is with Arbitrum One , Rinkeby and A. Rinkeby
 the only networks that are selectable are Ethereum and Gnosis when the wallet is not connected.
 
 Liquidity and Rewards are not disabled and can be clicked in the menu ✅ 
 user is able to connect to polygon (RPC endpoint is correct) ✅ 
 
**After connectin to Polygon:**

 Default currency is MATIC and balances are displayed correctly ✅ 
 
 Liquidity and Rewards are grayed out in the menu and can't be clicked. Badge 'Not available' should be visible
 Going to https://swapr.eth.link/#/pools or https://swapr.eth.link/#/rewards redirects to the landing page (routing is disabled for non-SWPR features) ✅ 
 
 Quickswap and Sushiswap are visible on ecoRouter when some common pair is selected ✅ 
 
There is still a bug that is found by Kinga regarding the token search and issue the was found by Kacper regarding the Bridge.

**Comment by adamazad** [comment]
> Polygon is visible but **is not selectable** in the Network Switcher ❌

How do I reproduce this?

**Comment by MilanVojnovic95** [comment]
> > Polygon is visible but **is not selectable** in the Network Switcher ❌
> 
> How do I reproduce this?
Open swapr and without connecting wallet try to select one of networks.

**Comment by denciu** [comment]
> There is a bug in token search (I believe that we had something like that before).
> 
> If you type "MA" in search "MATIC" appears (and it's correct behaviour) Screenshot 2022-04-28 at 14 59 08 But if you try search from "the middle" for example typing "TIC" MATIC will not appear (but wrapped MATIC will...) Screenshot 2022-04-28 at 15 00 34 That's quite strange behaviour.

It's not related to this PR, it's how Currency Search works in general. If you try to enter 'th' on mainnet you won't see ETH as well. If you think we should change this please make a separate issue.

**Comment by MilanVojnovic95** [comment]
Retested 
Polygon is visible and selectable in the Network Switcher ✅ 

Now we have problem with icons when connecting wallet:
image

There is still a bug that is found by Kinga and Daniel regarding the token search and issue the was found by Kacper regarding the Bridge.

https://user-images.githubusercontent.com/57331244/166872309-397806da-57a1-4af6-9915-b54b3cfdc0f9.mp4

File: src/components/Header/HeaderLink.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  border-radius: 3rem;
+  outline: none;
+  cursor: pointer;
+  text-decoration: none;
+  color: ${({ theme }) => theme.text5};
+  width: fit-content;
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 19.5px;
+  font-family: 'Montserrat';
```
Comment (Diogomartf) [inline src/components/Header/HeaderLink.tsx:20]: Do we have to declare the font-family here, don't we inherit from global styles?

File: src/components/Header/HeaderLink.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  outline: none;
+  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
+  text-decoration: none;
+  color: ${({ theme, disabled }) => (disabled ? transparentize(0.6, theme.text5) : theme.text5)};
+  font-weight: 400;
+  font-size: 14px;
+  line-height: 19.5px;
+  width: fit-content;
+  text-decoration: none !important;
+  font-family: 'Montserrat';
```
Comment (Diogomartf) [inline src/components/Header/HeaderLink.tsx:41]: Same here. Do we have to declare the font-family, don't we inherit from global styles?

File: src/components/Header/HeaderLink.tsx (hunk 2 above)
Comment (adamazad) [reply src/components/Header/HeaderLink.tsx:20]: Good point! @denciu just moved code. This PR shouldn't address this. Create a follow-up ticket for this.

**Comment by niemam29** [comment]
Retested and the issues that i mentioned above are still occuring

**Comment by adamazad** [comment]
> Retested and the issues that i mentioned above are still occuring

All of them?

**Comment by 0xVenky** [comment]
Looks good to me, Some of the issues that I found, I will create them as seperate issues and are not relevant for us to wait to merge.

**Comment by niemam29** [comment]
> > Retested and the issues that i mentioned above are still occuring
> 
> All of them?

Those one with network switcher and not opening swap page by default

File: src/components/Header/HeaderLink.tsx (hunk 3 above)
Comment (Diogomartf) [reply src/components/Header/HeaderLink.tsx:41]: Issue created 💪 

https://github.com/levelkdev/swapr-dapp/issues/948#issue-1226479243

**Comment by 0xVenky** [comment]
> 

Lets fix that after we merge as a seperate PR if possible?

**Comment by adamazad** [comment]
@niemam29, if you mean Polygon not being clickable on the Bridge page, that should be the behavior since EcoBridge does not support Polygon are the moment.

File: src/components/Header/HeaderLinkBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  left: 50%;
+  min-width: 100%;
+  transform: translate(-50%);
+`
+
+export interface HeaderLinkBadgeProps {
+  label: string
+}
+
+export const HeaderLinkBadge = ({ label }: HeaderLinkBadgeProps) => {
```
Comment (berteotti) [inline src/components/Header/HeaderLinkBadge.tsx:18]: Just out of curiosity, should we use `FC` type from React here?

File: src/components/Web3Status/ConnectWalletPopover.tsx [hunk 5]
```diff
@@ -128,7 +128,8 @@ export const ConnectWalletPopover = ({ setModal, tryActivation, children }: Conn
                 closeModals()
                 option.connector !== connector && !option.href && tryActivation(option.connector)
               }}
-              icon={require('../../assets/images/' + option.iconName)}
+              // eslint-disable-next-line
+              icon={require('../../assets/images/' + option.iconName).default}
```
Comment (berteotti) [inline src/components/Web3Status/ConnectWalletPopover.tsx:132]: I think that disabling eslint is a smell. Can we do this in another way?

**Review by berteotti — APPROVED** [review]
LGTM. Just made few comments that we can fix later.

**Review by 0xVenky — APPROVED** [review]
Looks good to be merged. There are some minor UI bugs that needs to be cleared before the release. But the change for supporting Polygon seems to work.

**Pull request merged**